readme.md

```diff
@@ -1,2 +1,282 @@
+# Job Application Backend API
+
 [![Code Quality Checks](https://github.com/ChrisThompsonK/team1-job-app-backend/actions/workflows/biome.yml/badge.svg)](https://github.com/ChrisThompsonK/team1-job-app-backend/actions/workflows/biome.yml)
-BackEnd ReadME
+
+A modern, type-safe Node.js backend API for managing job applications. Built with Express, TypeScript, and Drizzle ORM.
+
+## 🚀 Features
+
+- **Modern TypeScript**: Fully typed with ES modules
+- **Express.js**: Fast and minimal web framework
+- **Drizzle ORM**: Type-safe database operations with SQLite
+- **Advanced Filtering**: Server-side filtering with pagination and sorting
+- **Dependency Injection**: Clean architecture with IoC container
+- **Testing**: Comprehensive unit tests with Vitest
+- **Code Quality**: Biome for linting and formatting
+- **CORS Support**: Configurable cross-origin resource sharing
+- **Error Handling**: Centralized error handling middleware
+
+## 📋 Prerequisites
+
+- **Node.js** >= 18.0.0
+- **npm** or **yarn**
+
+## 🛠️ Installation
+
+1. Clone the repository:
+```bash
+git clone https://github.com/ChrisThompsonK/team1-job-app-backend.git
+cd team1-job-app-backend
+```
+
+2. Install dependencies:
+```bash
+npm install
+```
+
+3. Create a `.env` file in the root directory:
+```env
+PORT=3000
+NODE_ENV=development
+DATABASE_URL=file:./jobApp.db
+CORS_ORIGIN=http://localhost:3001
+```
+
+4. Set up the database:
+```bash
+npm run db:setup
+```
+
+This will run migrations and seed the database with initial data.
+
+## 🏃 Running the Application
+
+### Development Mode (with hot reload)
+```bash
+npm run dev
+```
+
+### Production Mode
+```bash
+npm run build
+npm start
+```
+
+The server will start on `http://localhost:3000` (or the PORT specified in your `.env` file).
+
+## 🧪 Testing
+
+```bash
+# Run tests in watch mode
+npm test
+
+# Run tests once
+npm run test:run
+
+# Run tests with coverage
+npm run test:coverage
+
+```
+
+## 🔍 Code Quality
+
+This project uses **Biome** for linting and formatting:
+
+```bash
+# Check for linting issues
+npm run lint
+
+# Fix linting issues automatically
+npm run lint:fix
+
+# Check formatting
+npm run format:check
+
+# Fix formatting
+npm run format
+
+# Run all checks (lint + format)
+npm run check
+
+# Fix all issues
+npm run check:fix
+```
+
+## 📚 API Endpoints
+
+### Base URL
+```
+http://localhost:3000
+```
+
+### Endpoints
+
+#### Get Welcome Message
+```
+GET /
+```
+
+#### Get All Jobs
+```
+GET /api/jobs
+```
+
+#### Get Job by ID
+```
+GET /api/jobs/:id
```

Comment:
it might be good to add an instructions file so that every time a new endpoint is added this file will be updated